compiler/src/iree/compiler/Codegen/Common/TileAndDistributeToWorkgroupsPass.cpp

```diff
@@ -247,12 +247,23 @@ struct LowerDispatchWorkgroupCountFromSetEncodingOp
   LowerDispatchWorkgroupCountFromSetEncodingOp(
       MLIRContext *context,
       IREE::LinalgExt::MaterializeEncodingInfo encodingInfo,
-      IREE::LinalgExt::MaterializeEncodingValueFn materializeEncodingValueFn,
-      RankedTensorType inputType, PatternBenefit benefit = 1)
+      RankedTensorType inputType, IREE::HAL::ExecutableTargetAttr targetAttr,
+      PatternBenefit benefit = 1)
       : OpRewritePattern(context, benefit),
         materializeEncodingInfo(std::move(encodingInfo)),
-        materializeEncodingValueFn(materializeEncodingValueFn),
-        inputType(inputType) {}
+        inputType(inputType) {
+    if (isVMVXBackend(targetAttr) && hasMicrokernels(targetAttr)) {
```

Comment:
We can drop this condition. Same problem should happen on any backend anyway.

Reply:
But without knowing that we are on VMVX+ukernel, we can't know what is a reasonable default tile size value?  Here, 16 is OK because we know that vmvx+ukernel actual tile size values are factors of 16?

Reply:
we have this assumption upstream when we pad to 16 assuming it is enough for all backends. So its consistent with that mostly.

Reply:
ok